apps/nextjs-app-router/src/app/[lang]/[[...path]]/page.tsx

```diff
@@ -2,20 +2,39 @@ import { client } from '@/makeswift/client'
 import '@/makeswift/components'
 import { getSiteVersion } from '@makeswift/runtime/next/server'
 import { notFound } from 'next/navigation'
-import { Page as MakeswiftPage } from '@makeswift/runtime/next'
-import { locales } from '@/localization'
+import {
+  Page as MakeswiftPage,
+  MakeswiftPageResult,
+} from '@makeswift/runtime/next'
 
 type ParsedUrlQuery = { lang: string; path?: string[] }
 
 export async function generateStaticParams() {
-  const pages = await client.getPages()
+  let hasMorePages = true
+  let after: string | undefined = undefined
+  const pages: MakeswiftPageResult[] = []
 
-  return pages.flatMap((page) =>
-    locales.map((locale) => ({
-      path: page.path.split('/').filter((segment) => segment !== ''),
-      lang: locale,
+  do {
+    const { data, hasMore } = await client.getPages({ after })
+    pages.push(...data)
+    hasMorePages = hasMore
+    after = data.at(-1)?.id
+  } while (hasMorePages)
+
+  return pages.flatMap((page) => [
+    {
+      params: {
+        path: page.path.split('/').filter((segment) => segment !== ''),
+        lang: page.locale,
+      },
+    },
+    ...page.localizedVariants.map((variant) => ({
+      params: {
+        path: variant.path.split('/').filter((segment) => segment !== ''),
+        lang: variant.locale,
+      },
```

Comment:
I think this is better since there could be certain variants that are offline and so the cartesian product might include 404s that we'd otherwise not want